### DISCUSSION TITLE
Run npm run export automatically on editing template file in GitHub directly

### DISCUSSION THREAD

**Discussion by anjanesh** [community/community#28726] [Codespaces] [answered]
Is there a way to do this ?

I have a NextJS app on GitHub with 
`"export": "npm run build && next export -o _static"`
DigitOcean App Platform looks for **_static** folder in NextJS for a static-only app. [Reference](https://docs.digitalocean.com/tutorials/app-nextjs-deploy/).

If my team member edits HTML content in pages/about.tsx then I want GitHub to run `npm run export` so that the **_static** folder is automatically updated for the site's recent changes to be reflected in DigitalOcean.

**Comment by airtower-luna** [comment, marked as answer]
Take a look at [Actions](https://docs.github.com/en/actions/quickstart), and set up a workflow that does the build and deployment. Edits over the web UI will give you a push event just like pushing from your computer does. :slightly_smiling_face:

**Reply by anjanesh** [reply]
So the same 
```
on:
  push
```
is the same for web UI edits as well ?

**Reply by airtower-luna** [reply]
Yes, exactly. A web edit adds a commit to a branch, so it's a `push` event. And if the commit goes through a PR you'll also get `pull_request`, and another `push` on the target branch when the PR is merged.

If you deploy from the repository I assume you'll want to have a branch filter for deployment, so commits to different branches don't overwrite each other. :wink:

**Reply by anjanesh** [reply]
Thanks a bunch.
